Gems/LmbrCentral/Code/Source/Builders/CopyDependencyBuilder/FontBuilderWorker/FontBuilderWorker.cpp

```diff
@@ -52,7 +51,8 @@ namespace CopyDependencyBuilder
         
         if (fileExtension == "font" || fileExtension == "fontfamily")
         {
-            return azrtti_typeid<LyShine::FontAsset>();
+            static AZ::Data::AssetType fontAssetType("{57767D37-0EBE-43BE-8F60-AB36D2056EF8}"); // form UiAssetTypes.h in the LyShine gem
+            return fontAssetType;
```

Comment:
Yeah, its not wrong to do this if you want to avoid using headers in here.

This is a result of "Copy Dependency Builder" being a bit of a catch-all, to scan dependencies of legacy files that had no builder of their own and did not emit dependencies.

The actual correct way to do this would be for lyshine itself (as in the lyshine gem) to have a builder module (or use the editor module in builder), which knows how to express its dependencies, and issues the copy jobs itself and exports its dependencies as part of its processing job.

Reply:
The other interesting thing here is that font assets are not exclusive to LyShine so the font asset type and copy builder would probably go into a more appropriate gem rather than LyShine.